### DISCUSSION TITLE
stuck in "The volume is in expansion progress from size 20Gi to 10Gi"

### DISCUSSION THREAD

**Discussion by usma0118** [longhorn/longhorn#4459] [Q&A]
No changes made to pod or longhorn, one of volumes is stuck in volume expansion in progress.

![image](https://user-images.githubusercontent.com/355154/185870007-4691b1e7-1701-4b9e-b96a-de2c4e04c793.png)

Any ideas how to fix this?

have tried scaling down workload to 0 and reattaching volume... no changes.

**Comment by derekbit** [comment]
Can you help check the longhorn-manager and instance-manager logs to see if there is any clue?

**Reply by usma0118** [reply]
Found:

Volume size was reduced.

`failed to update status for engine pvc-604aa686-a9de-45ca-9b97-169ec8db2042-e-8aad8b64: BUG: The expected size 10737418240 of engine pvc-604aa686-a9de-45ca-9b97-169ec8db2042-e-8aad8b64 should not be smaller than the current size 21474836480`

**Reply by innobead** [reply]
@smallteeths Just found we did not use `IsExpending` of volume controller but rather decide the volume in expanding state if the size is different between spec and status. This would cause some misunderstanding. We should probably improve this. 

https://github.com/longhorn/longhorn-ui/blob/67bf11df68a06fbbf7fed9082acfd6498edfd3c7/src/routes/volume/VolumeList.js#L187-L189

**Reply by smallteeths** [reply]
Ok, I'm not quite sure if the front end can get the error. If I can get it, I'm going to improve this.  Use the tooltip shows the error when the volume expend fails.

**Reply by usma0118** [reply]
tooltip will help.

**Reply by shuo-wu** [reply]
The field `v.controllers[0].expanding` exposed to the frontend is the same as `e.Status.IsExpanding`. But `e.Status.IsExpanding` [may be not updated](https://github.com/longhorn/longhorn-manager/blob/76f0fb4badcc8e0f5baf45e3b739767d0971ed6b/controller/engine_controller.go#L841) since the expansion is quite quick. I am thinking if we need to remove it. cc @innobead

**Reply by innobead** [reply]
It seems like eventually consistent, so I don't think we should remove this. But @smallteeths will have some tweaks for UI to avoid any confusion.
